### DISCUSSION TITLE
How to handle software with paid major version upgrades?

### DISCUSSION THREAD

**Discussion by ThioJoe** [microsoft/winget-pkgs#16341] [Q&A]
I was thinking about how some companies release a new major version of software every year, but current owners would need to pay to upgrade to the latest version. For example, Camtasia 2020 vs 2021 or whatever.

**Should this software maybe have each 'year' / major version get it's own separate entry?** Otherwise, if someone doesn't want to pay for a new version, they'll have to constantly see an upgrade available.

Maybe it's semantics, but one could argue that if a paid upgrade is required to buy a new 'version', it could be considered a totally independent piece of software. Or as a thought, perhaps in the future, there could be an option to ignore upgrades for higher major versions?

**Comment by denelon** [comment]
@ThioJoe we've encountered a broad range of behaviors for the way various installers report their versions, and how publishers / Independent Software Vendors (ISVs) report software versions (IntelliJ IDEA is a great example for this). I'm not sure there is a one size fits all here. We've been discussing some of the differences with respect to [package pinning](https://github.com/microsoft/winget-cli/issues/476) and the current behavior for several programming languages to present "major" versions as different packages (Java is a great example for this).

We've also seen some suggestions related to how the "License" could be used. I think one of our bigger challenges is related to the nature of a community submitted package as opposed to one published by a "verified publisher".

This is a great topic to have for the community as well as software publishers and ISVs.
